src/realm/sync/client.cpp

```diff
@@ -1797,20 +1829,28 @@ void SessionWrapper::report_progress(bool only_if_new_uploadable_data)
     // uploadable_bytes is uploaded + remaining to upload, while downloadable_bytes
     // is only the remaining to download. This is confusing, so make them use
     // the same units.
-    std::uint_fast64_t total_bytes = downloaded_bytes + downloadable_bytes;
+    downloadable_bytes += downloaded_bytes;
 
     m_sess->logger.debug("Progress handler called, downloaded = %1, "
-                         "downloadable(total) = %2, uploaded = %3, "
-                         "uploadable = %4, reliable_download_progress = %5, "
-                         "snapshot version = %6",
-                         downloaded_bytes, total_bytes, uploaded_bytes, uploadable_bytes,
-                         m_reliable_download_progress, snapshot_version);
-
-    // FIXME: Why is this boolean status communicated to the application as
-    // a 64-bit integer? Also, the name `progress_version` is confusing.
-    std::uint_fast64_t progress_version = (m_reliable_download_progress ? 1 : 0);
-    m_progress_handler(downloaded_bytes, total_bytes, uploaded_bytes, uploadable_bytes, progress_version,
-                       snapshot_version);
+                         "downloadable = %2, uploaded = %3, "
+                         "uploadable = %4, snapshot version = %5",
+                         downloaded_bytes, downloadable_bytes, uploaded_bytes, uploadable_bytes, snapshot_version);
+
+    // FIXME take into account first initial reported value in this batch
+    double upload_estimate = uploadable_bytes > 0 ? (uploaded_bytes / double(uploadable_bytes)) : 1;
+    double download_estimate = downloadable_bytes > 0 ? (downloaded_bytes / double(downloadable_bytes)) : 1;
+
+    if (m_download_estimate) {
+        download_estimate = m_download_estimate.value();
+        if (m_transient_downloaded_bytes)
+            downloaded_bytes += m_transient_downloaded_bytes.value();
+
+        // FIXME add some estimate to this
```

Comment:
I think for FLX we can't estimate this and the download estimate is kinda what you get in lieu of downloadable bytes.

Reply:
yes, but i've added at least for now some logic to it, until we fully deprecate pbs and remove this api. That's what was desired from sdks which will still try non breaking release first.